components/eam/src/physics/cam/hetfrz_classnuc_cam.F90

```diff
@@ -539,114 +513,129 @@ end subroutine hetfrz_classnuc_cam_init
 
 !================================================================================================
 
-subroutine hetfrz_classnuc_cam_calc( &
-   state, deltatin, factnum, pbuf)
+subroutine hetfrz_classnuc_cam_calc( ncol, lchnk, temperature, pmid, rho, ast, &
+                                     qc, nc, state_q, aer_cb, deltatin, factnum, &
+                                     frzimm, frzcnt, frzdep)
```

Comment:
Thanks for teasing out the intent in/outs for the subroutine. Now it is very clear which variables are in and which are out. Would you please also specify in-outs for these variables here?